address-space-controller/src/main/java/io/enmasse/controller/StatusController.java

```diff
@@ -39,15 +40,16 @@ public StatusController(Kubernetes kubernetes, SchemaProvider schemaProvider, In
     }
 
     @Override
-    public AddressSpace reconcileActive(AddressSpace addressSpace) {
+    public AddressSpace reconcileActive(AddressSpace addressSpace) throws IOException {
         if (addressSpace.getStatus().isReady()) {
             checkComponentsReady(addressSpace);
             checkAuthServiceReady(addressSpace);
             checkExposedEndpoints(addressSpace);
         }
 
         if (addressSpace.getStatus().isReady()) {
-            if (addressSpace.getSpec().getPlan().equals(addressSpace.getAnnotation(AnnotationKeys.APPLIED_PLAN))) {
+            AddressSpaceSpec appliedSpec = AppliedConfig.parseCurrentAppliedConfig(addressSpace);
+            if (addressSpace.getSpec().equals(appliedSpec)) {
                 addressSpace.getStatus().setPhase(Phase.Active);
             }
         } else {
```

Comment:
On changing the authentication-service of an existing brokered addressspace, I saw the addressspace momentarily go through the Failed state.  This seems wrong

Reply:
I think that is because when configuration is applied, the phase is set to Configuring, but the StatusController status checks may set it back to active immediately if the changes have not caused readiness to fail yet. Then in the next iteration of the controller loop, it would be possible for statuscontroller to set it to Failed if the status check fails.

I think there should probably be a threshold of setting status to Failed based on some configurable time interval. I.e. status checks failed for 2 minutes. I'd might prefer to address this in a separate PR though.